flink-clients/src/test/java/org/apache/flink/client/ExecutionEnvironmentTest.java

```diff
@@ -91,9 +106,42 @@ public void testUserDefinedJobNameWithConfigure() {
         testJobName(jobName, env);
     }
 
+    @Test
+    public void testConfigureFileSystem() throws Exception {
+        final Configuration config = new Configuration();
+        EXPECTED_CONFIGURATION.forEach(config::setString);
+        final StreamExecutionEnvironment environment = new LocalStreamEnvironment(config);
+        environment.getCheckpointConfig().setCheckpointStorage("testConfig:/" + tmp.newFolder());
+        environment.enableCheckpointing(100);
+        environment.fromElements(1, 2, 3, 4).map((MapFunction<Integer, Integer>) value -> value);
+        // We only want to enable the configuration check for this test case. Nevertheless, the
+        // filesystem is loaded and configured for all test cases in this module.
+        expectConfiguration = true;
```

Comment:
Can't you use `SharedReference` here?

Reply:
How would I pass the `SharedReference` to the filesystem since it is loaded through SPI? So far I only used `SharedReference` for cases where I could somehow inject the variable.

Reply:
Fair point. 

On a related note: Does that configuration really need to be volatile?